frame/ethereum/src/lib.rs

```diff
@@ -219,12 +219,18 @@ impl<T: Config> frame_support::unsigned::ValidateUnsigned for Module<T> {
 				return InvalidTransaction::Payment.into();
 			}
 
-			if transaction.gas_price < T::FeeCalculator::min_gas_price() {
+			let min_gas_price = T::FeeCalculator::min_gas_price();
+
+			if transaction.gas_price < min_gas_price {
```

Comment:
This is fine for now, largely because `T::FeeCalculator::min_gas_price()` is `0`, but going forward (with either `pallet_dynamic_fee` or EIP-1559) this could be a value that changes from block to block.

Some questions I had about the way this code is called:

1) Will this reject a TXN submitted to an RPC node?
2) Will this purge a TXN from the mempool?

In the future, there will be the idea of "this TXN wouldn't be valid in this block, but might in the future" (much like a `txn.nonce > account.nonce`). This appears to be handled above by returning `return InvalidTransaction::Stale.into();`...

Reply:
> Will this reject a TXN submitted to an RPC node

Yes, returning `InvalidTransaction` here propagates back to the rpc handler and the response to the original request will be a jsonrpc standard error -32603.

> Will this purge a TXN from the mempool?

Not sure I understand this question, you mean on overriding (same nonce)?

> In the future, there will be the idea of "this TXN wouldn't be valid in this block, but might in the future" 

This (transactions that go to the future queue of the transaction pool) is already handled with `and_provides` and `and_requires` in the `ValidTransactionBuilder`.